example/resnet/resnet.go

```diff
@@ -63,61 +65,29 @@ func accuracy(output, target torch.Tensor, topk []int64) []float32 {
 	return res
 }
 
-func imageNetLoader(tarFile string, batchSize int) (*datasets.ImageNetLoader, error) {
-	f, e := os.Open(tarFile)
-	if e != nil {
-		panic(e)
-	}
-	vocab, e := datasets.BuildLabelVocabulary(f)
-	fmt.Println("building label vocabulary done.")
-	if e != nil {
-		return nil, e
-	}
-	if _, e := f.Seek(0, io.SeekStart); e != nil {
-		return nil, e
-	}
+func imageNetLoader(r io.Reader, vocab map[string]int64, batchSize int, skipSamples int) (*datasets.ImageNetLoader, error) {
 	trans := transforms.Compose(
 		transforms.RandomResizedCrop(224),
 		transforms.RandomHorizontalFlip(0.5),
 		transforms.ToTensor(),
 		transforms.Normalize([]float64{0.485, 0.456, 0.406}, []float64{0.229, 0.224, 0.225}))
 
-	loader, e := datasets.ImageNet(f, vocab, trans, batchSize)
+	loader, e := datasets.ImageNet(r, vocab, trans, batchSize, skipSamples)
 	if e != nil {
 		return nil, e
 	}
 	return loader, nil
 }
 
-func train(model *models.ResnetModule, opt torch.Optimizer, batchSize int, device torch.Device, tarFile string) {
-	model.Train(true)
-	loader, e := imageNetLoader(tarFile, batchSize)
-	if e != nil {
-		panic(e)
-	}
-	batchIdx := 1
-	startTime := time.Now()
-	for loader.Scan() {
-		image, target := loader.Minibatch()
-		image = image.To(device, torch.Float)
-		target = target.To(device, torch.Long)
-		output := model.Forward(image)
-		loss := F.CrossEntropy(output, target, torch.Tensor{}, -100, "mean")
-
-		acc := accuracy(output, target, []int64{1, 5})
-		acc1 := acc[0]
-		acc5 := acc[1]
-		if batchIdx%100 == 0 {
-			throughput := float64(100*batchSize) / time.Since(startTime).Seconds()
-			fmt.Printf("batch: %d, loss: %f, acc1 :%f, acc5: %f, throughput: %.2f samples/sec\n", batchIdx, loss.Item(), acc1, acc5, throughput)
-		}
-
-		opt.ZeroGrad()
-		loss.Backward()
-		opt.Step()
-		batchIdx++
-	}
-	torch.FinishGC()
+func trainOneBatch(image, target torch.Tensor, model *models.ResnetModule, opt torch.Optimizer) (float32, float32, float32) {
```

Comment:
In deep learning, trainOneBatch is known as an iteration. Also, it is minibatch, not batch.